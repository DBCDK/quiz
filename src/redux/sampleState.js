import Immutable from 'immutable';
import {quizData} from '../quizData.js';

export const sampleState = Immutable.fromJS({
  admin: {
    // edit a single screen if currentScreen is set
    currentScreen: undefined
  },
  widget: {
    // will be set to true, when all data is loaded
    initialised: false,
    // non-zero if loading, increases with each load operation, and decreases when done
    loading: 0,
    currentScreen: undefined
  },
  quizState: {},

  // data for the current Quiz
  quiz: undefined,

  searchQuery: {
    query: '',
    ownOnly: true
  },

  searchResults: [],

  storage: {
    // storage user id, initialised on start
    user: undefined,

    // storage type uuids, initialised on start
    quizType: undefined,
    imageType: undefined,

    // mirror of data in storage, used for to find local changes to sync to storage
    quiz: {},
    page: {}
  }
});
export const newQuiz = sampleState.set('quiz', Immutable.fromJS(quizData()));
