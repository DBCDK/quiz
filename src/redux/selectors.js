export const ownQuizzes = state =>
  state.getIn(['ui', 'ownQuizzes']).map(uuid => state.getIn(['quiz', uuid]));
export const quizVariables = state => state.get('quizState');
export const currentScreen = state =>
  state.getIn(['quiz', 'screens', state.getIn(['ui', 'currentScreen'])], []);
export const quizDescription = state => state.getIn(['quiz', 'description']);
export const loading = state => state.getIn(['ui', 'loading']) !== 0;
