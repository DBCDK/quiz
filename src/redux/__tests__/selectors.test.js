import {ownQuizzes, loading, questionList} from '../selectors';
import sampleState from '../sampleState';
import Immutable from 'immutable';

describe('selectors', () => {
  describe('ownQuizzes', () => {
    it('finds quizzes', async () => {
      expect(
        ownQuizzes(
          Immutable.fromJS({
            ui: {ownQuizzes: ['a', 'b']},
            quiz: {a: 'c', b: 'd'}
          })
        ).toJS()
      ).toEqual(['c', 'd']);
    });
  });
  describe('loading', async () => {
    it('returns false when not loading', async () => {
      expect(loading(Immutable.fromJS({ui: {loading: 0}}))).toBe(false);
    });
    it('returns true when loading', async () => {
      expect(loading(Immutable.fromJS({ui: {loading: 2}}))).toBe(true);
    });
  });
});
describe('admin selectors', () => {
  describe('topNodes', () => {
    it('finds top nodes screen graph', () => {
      expect(questionList(sampleState)).toMatchSnapshot();
    });
  });
});
