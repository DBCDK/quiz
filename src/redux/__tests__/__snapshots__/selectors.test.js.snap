// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin selectors topNodes finds top nodes screen graph 1`] = `
Array [
  "intro",
  "question1",
  "question2",
  "done",
]
`;
